packages/hash/api/src/graphql/resolvers/linkedAggregation/createLinkedAggregation.ts

```diff
@@ -5,18 +5,17 @@ import {
 } from "../../apiTypes.gen";
 import { Entity, UnresolvedGQLLinkedAggregation } from "../../../model";
 import { LoggedInGraphQLContext } from "../../context";
+import { DBAdapter } from "../../../db";
+import __User from "../../../model/user.model";
 
-export const createLinkedAggregation: Resolver<
-  Promise<UnresolvedGQLLinkedAggregation>,
-  {},
-  LoggedInGraphQLContext,
-  MutationCreateLinkedAggregationArgs
-> = async (
-  _,
-  { sourceAccountId, sourceEntityId, path, operation },
-  { dataSources, user },
-) =>
-  dataSources.db.transaction(async (client) => {
+export const createLinkedAggregationRaw: (
+  args: MutationCreateLinkedAggregationArgs,
+  config: { db: DBAdapter; user: __User },
+) => Promise<UnresolvedGQLLinkedAggregation> = async (args, config) => {
+  const { operation, path, sourceAccountId, sourceEntityId } = args;
+  const { db, user } = config;
+
+  return db.transaction(async (client) => {
```

Comment:
What's this for?

Reply:
This is used inside `mockData/index.ts` for creating a new linked aggregation.

Reply:
@akash-joshi is there a way we can just use the model classes directly instead?

Reply:
I will attempt this.

Reply:
@benwerner01 actually the method being implemented here is used both within the resolver, and outside, in the mockData file.

If I use models there, that will result in code duplication. Is that fine?

Reply:
The easiest thing here is probably to use the resolver directly, and just assert that the object you're passing in the context is the one that's expected. Saves duplication. There's a bit of type unsafety introduced but I think that's fine for creating mock data.

Reply:
> If I use models there, that will result in code duplication. Is that fine?
@akash-joshi if I'm understanding correctly, and you're talking about "code duplication" of this:
```TypeScript
    const source = await Entity.getEntityLatestVersion(client, {
      accountId: sourceAccountId,
      entityId: sourceEntityId,
    });

    ...

    const aggregation = await source.createAggregation(client, {
      stringifiedPath: path,
      operation: { .... },
      createdBy: user,
    });
```
that's totally fine in my opinion, and I don't think I'd even consider this code duplication.